test/hotspot/jtreg/compiler/arraycopy/stress/AbstractStressArrayCopy.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2021, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+import java.util.Random;
+import jdk.test.lib.Utils;
+
+public abstract class AbstractStressArrayCopy {
+    /**
+     * Max array size to test.
+     */
+    static final int MAX_SIZE = 1024*1024 + 1;
```

Comment:
Do we really need such big arrays for regression testing. It may make sense for JMH but not for these tests I think.

Reply:
My original intent was to make sure the tests cross all small page sizes (up to 64K) and maybe even some large page sizes (1M `long[]` is 8M, so 2*4M). The size of this array does not matter for test performance very much, since we only allocate two `MAX_SIZE`-d arrays per entire run. Driver even caps the heap size at `-Xmx256m` to block tests from using too much memory. So, I'd leave it at 1M, if you agree.

Reply:
Okay. I was concern because of times you show for 
# x86_64 (i5-11500)
  real    41m32.622s
  user    447m19.986s
  sys     0m21.026s
```

Reply:
That small machine has very slow memory compared to other ones. The parallelism in stress tests (9 types, 2 forked VMs each) puts that machine on its knees. There is a blurb about that effect here: https://github.com/openjdk/jdk/pull/6594/files#diff-f72fee20a49daaf4e05002372e93f426407ecd429a227393e2ec79e821042c90R40-R47 -- I don't think it would matter much if we trim `MAX_SIZE`, but I'll try tomorrow.

Reply:
All right. `MAX_SIZE` actually makes a lot of difference for that machine. I trimmed it down to 128K to cater for 64K pages, and added some explanation for the choice. See new commit. Also updated the PR body with new timings.